services/api/service.go

```diff
@@ -421,12 +422,25 @@ func (api *RelayAPI) StartServer() (err error) {
 		}
 	}
 
-	if !foundCapellaEpoch || !foundDenebEpoch {
-		return ErrMissingForkVersions
+	// feature flags
+	ffCheckDenebForkSchedule := false // compatibility with older CL clients missing deneb fork schedule
+	if os.Getenv("CHECK_DENEB_FORK_SCHEDULE") != "" {
+		log.Warn("env: CHECK_DENEB_FORK_SCHEDULE: checking validation for deneb fork schedule")
+		ffCheckDenebForkSchedule = true
+	}
+
+	if ffCheckDenebForkSchedule {
+		if !foundCapellaEpoch || !foundDenebEpoch {
+			return ErrMissingForkVersions
+		}
+	} else {
+		if !foundCapellaEpoch {
+			return ErrMissingForkVersions
+		}
 	}
 
 	// Print fork version information
-	if hasReachedFork(currentSlot, api.denebEpoch) {
+	if ffCheckDenebForkSchedule && hasReachedFork(currentSlot, api.denebEpoch) {
```

Comment:
couldn't we replace this with a check if `api.denebEpoch == 0`?

```suggestion
	if api.denebEpoch != 0 && hasReachedFork(currentSlot, api.denebEpoch) {
```

Reply:
there are two cases in which `api.denebEpoch = 0`
1. The beacon node is outdated and did not return a Deneb fork schedule in which case `api.denebEpoch` is never assigned and `foundDenebEpoch = false`
2. The beacon node returned a 0 as the epoch for Deneb and thus `foundDenebEpoch = true`. 
We can not check `api.denebEpoch != 0` alone as due to the feature flag it is possible `api.denebEpoch` was never assigned. The most correct way is to check is `foundDenebEpoch && hasReachedFork(currentSlot, api.denebEpoch)`